src/components/a-checkbox/_preview.html

```diff
@@ -1,2 +1,32 @@
-<axa-checkbox id="test" value="a" name="test-name" ></axa-checkbox>
-<label for="test">hello</label>
+<axa-checkbox
+  input-id="checkbox"
+  value="a"
+  name="test-name2"
+>
+  hello
+</axa-checkbox>
+<axa-checkbox
+  input-id="checkbox1"
```

Comment:
Please make id optional and as default use the UUID generator. If a consumer copies this code, he will copy the id name and therefore for more than one element we will have the same ID and therefore the HTML is invalid.

so please remove input-id in the preview.html s and add something like this in the template:

```
import html from 'nanohtml';
import BaseComponent from '../../abstarct/base-component';

export default function ({
  inputId = BaseComponent.uuid(),
  value,
  name,
  checked = false,
  disabled = false,
}, childrenFragment) {
  return html`<label class="a-radio__wrapper">
    <input
      id="${inputId}"
      class="a-radio__input"
      type="radio"
      name="${name}"
      checked="${checked}"
      value="${value}"
      disabled="${disabled}">
    <span class="a-radio__icon"></span>${childrenFragment}
  </label>`;
}

```

In this way, only in EXEPTIONAL cases id will be set from outside and we have just to trust that the user knows HTML

Reply:
but it is still `id="${inputId}"` here on the <input> tag and that is my problem

Reply:
People will copy paste html for the WC so in my opinion better to have a defaul UUID ID and only if needed set the id outside

Reply:
having same id on input field is one of the most common errors done in HTML programming due to copy pasta and to "but it works so what".

Reply:
Sorry, my bad I missread the code section. Your are absolutely right.